### PR TITLE
ps1encode: added new package per issue #253

### PR DESCRIPTION
**Pull request by strayArch**

### CODE REVIEW

File: archstrike/ps1encode/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  install -dm755 "${pkgdir}/usr/share/${_pkgname}"
+  install -Dm644 LICENSE "${pkgdir}/usr/share/licenses/${_pkgname}/LICENSE"
+  install -Dm755 "${_pkgname}.rb"  "${pkgdir}/usr/share/${_pkgname}/"
+  # Create script to call it's doppelgaenger
+  cat > "${pkgdir}/usr/bin/${_pkgname}" << EOF
+#!/usr/bin/env bash
+cd "/usr/share/${_pkgname}"
+ruby ${_pkgname}.rb "\$@"
+EOF
+  chmod +x "${pkgdir}/usr/bin/${_pkgname}"
```
Comment (xorond) [inline archstrike/ps1encode/PKGBUILD]: chmod 755

File: archstrike/ps1encode/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Maintainer: ArchStrike <team@archstrike.org>
+# Contributor: strayArch <j.a.stronz@gmail.com>
+
+buildarch=1
+
+_pkgname=ps1encode
+pkgname=${_pkgname}-git
```
Comment (xorond) [inline archstrike/ps1encode/PKGBUILD]: if pkgname ps1encode-git, it needs to be in folder `archstrike/ps1encode-git`
i'd also prefer if you put it into `archstrike-testing` instead

File: archstrike/ps1encode/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+pkgver() {
+  cd "${_pkgname}"
+  printf "r%s.%s" "$(git rev-list --count HEAD)" "$(git rev-parse --short HEAD)"
+}
+
+package() {
+  cd "${_pkgname}"
+  install -dm755 "${pkgdir}/usr/bin"
+  install -dm755 "${pkgdir}/usr/share/${_pkgname}"
```
Comment (xorond) [inline archstrike/ps1encode/PKGBUILD]: change all instances of `/usr/share/${_pkgname}` to `/usr/share/${pkgname}`

File: archstrike/ps1encode/PKGBUILD [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+pkgver=r41.68d7778
+pkgrel=1
+pkgdesc="A tool to generate and encode a PowerShell based Metasploit payloads"
+url="https://github.com/CroweCybersecurity/ps1encode"
+arch=('any')
+groups=('archstrike' 'archstrike-exploit' 'archstrike-malware')
+license=('MIT')
+depends=('ruby' 'metasploit')
+makedepends=('git')
+source=("${_pkgname}::git+https://github.com/CroweCybersecurity/${_pkgname}.git")
```
Comment (xorond) [inline archstrike/ps1encode/PKGBUILD]: `source=("${_pkgname}::git+${url}.git")`

**Comment by xorond** [comment]
add dependencies:
- [ ] community/mingw-w64-gcc

File: archstrike/ps1encode/PKGBUILD (hunk 2 above)
Comment (strayArch) [inline archstrike/ps1encode/PKGBUILD]: oops. done and done

File: archstrike-testing/ps1encode-git/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+_pkgname=ps1encode
+pkgname=${_pkgname}-git
+pkgver=r41.68d7778
+pkgrel=1
+pkgdesc="A tool to generate and encode a PowerShell based Metasploit payloads"
+url="https://github.com/CroweCybersecurity/ps1encode"
+arch=('any')
+groups=('archstrike' 'archstrike-exploit' 'archstrike-malware')
+license=('MIT')
+depends=('ruby' 'metasploit' 'mingw-w64-gcc')
```
Comment (xorond) [inline archstrike-testing/ps1encode-git/PKGBUILD]: let's put `mingw-w64-gcc` in `optdepends` and it should be good to go

**Pull request merged**